ISL.ReIdentification.Core.Tests.Unit/Services/Foundations/UserAccesses/UserAccessesTests.cs

```diff
@@ -165,5 +183,43 @@ private static Expression<Func<Xeption, bool>> SameExceptionAs(
             return actualException =>
                 actualException.SameExceptionAs(expectedException);
         }
+
+        public static TheoryData<List<UserAccess>, List<OdsData>, List<string>> OrganisationListsAndExpectedOutputs()
+        {
+            UserAccess futureActiveFromUserAccess = CreateRandomUserAccess(GetRandomFutureDateTimeOffset());
+            UserAccess pastActiveToUserAccess = CreateRandomUserAccess(GetRandomPastDateTimeOffset());
+            UserAccess currentActiveToUserAccess = CreateRandomUserAccess();
+            currentActiveToUserAccess.ActiveFrom = DateTimeOffset.UtcNow;
+            currentActiveToUserAccess.ActiveTo = GetRandomFutureDateTimeOffset();
+
+            List<UserAccess> userAccesses = new List<UserAccess> {
+                futureActiveFromUserAccess,
+                pastActiveToUserAccess,
+                currentActiveToUserAccess
+            };
+
+            List<string> randomUserOrganisations = new List<string> { currentActiveToUserAccess.OrgCode };
+
+            List<OdsData> randomOdsDataItems = CreateRandomOdsDatas(randomUserOrganisations);
+            List<OdsData> storageOdsDataItems = randomOdsDataItems.DeepClone();
+            List<string> allUserOrganisation = randomUserOrganisations;
+
+            foreach (OdsData odsData in randomOdsDataItems)
+            {
+                List<OdsData> randomOdsDatas = CreateRandomOdsDataChildren(odsData.OdsHierarchy);
+                foreach (OdsData randomOdsData in randomOdsDatas)
+                {
+                    randomOdsData.RelationshipWithParentEndDate = GetRandomPastDateTimeOffset();
+                }
+                storageOdsDataItems.AddRange(randomOdsDatas);
+            }
```

Comment:
foreach is a multiline code block, so need a new line before and after the foreach

```suggestion
            foreach (OdsData odsData in randomOdsDataItems)
            {
                List<OdsData> randomOdsDatas = CreateRandomOdsDataChildren(odsData.OdsHierarchy);
                
                foreach (OdsData randomOdsData in randomOdsDatas)
                {
                    randomOdsData.RelationshipWithParentEndDate = GetRandomPastDateTimeOffset();
                }
                
                storageOdsDataItems.AddRange(randomOdsDatas);
            }
```